applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1015,6 +1021,36 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateAll(MatrixType&        rLe
     KRATOS_CATCH("")
 }
 
+template <unsigned int TDim, unsigned int TNumNodes>
+std::vector<double> UPwSmallStrainElement<TDim, TNumNodes>::CalculateDerivativesOfSaturation(
+    const std::vector<double>& fluid_pressures, RetentionLaw::Parameters& RetentionParameters)
+{
+    std::vector<double> result;
+    std::transform(fluid_pressures.begin(), fluid_pressures.end(), mRetentionLawVector.begin(),
+                   std::back_inserter(result),
+                   [&RetentionParameters](double fluid_pressure, RetentionLaw::Pointer pRetentionLaw) {
+        RetentionParameters.SetFluidPressure(fluid_pressure);
+        return pRetentionLaw->CalculateDerivativeOfSaturation(RetentionParameters);
+    });
+
+    return result;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+std::vector<double> UPwSmallStrainElement<TDim, TNumNodes>::CalculateDegreesOfSaturation(
+    const std::vector<double>& fluid_pressures, RetentionLaw::Parameters& RetentionParameters)
```

Comment:
Done, as per your suggestion I removed the `GeoTransportEquationUtilities::CalculateDegreesSaturation` and used the new function instead. Although there are then 2 versions of the function, we hope this duplication will be removed soon by merging the diff-order/non-diff order branch.

FYI @markelov208 since you separated out the utility. Please let me know if you have any reservations with this approach